src/Features/LanguageServer/Protocol/Handler/Diagnostics/Public/PublicWorkspacePullDiagnosticsHandler_IOnInitialized.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.SolutionCrawler;
+using Roslyn.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Diagnostics.Public
+{
+    internal sealed partial class PublicWorkspacePullDiagnosticsHandler : IOnInitialized
+    {
+        public async Task OnInitializedAsync(ClientCapabilities clientCapabilities, RequestContext context, CancellationToken cancellationToken)
+        {
+            if (clientCapabilities?.TextDocument?.Diagnostic?.DynamicRegistration is true && IsFsaEnabled())
+            {
+                var sources = _diagnosticSourceManager.GetSourceNames(isDocument: false);
+                var regParams = new RegistrationParams
+                {
+                    Registrations = sources.Select(FromSourceName).ToArray()
+                };
+                await _clientLanguageServerManager.SendRequestAsync(
+                    methodName: Methods.ClientRegisterCapabilityName,
+                    @params: regParams,
+                    cancellationToken).ConfigureAwait(false);
+
+                Registration FromSourceName(string sourceName)
+                {
+                    return new()
+                    {
+                        Id = sourceName,
+                        Method = Methods.WorkspaceDiagnosticName,
```

Comment:
I think this still fails right?  
Due to https://github.com/microsoft/language-server-protocol/issues/1723#issuecomment-2071155812 this needs to be registered against the textdocument method name, see https://github.com/dibarbet/roslyn/commit/4b329cd5235ed3d54fed3925e2666eca6ebd7823

When fixed, also add a comment linking back to the language-server-protocol github issue